packages/atomic/src/components/facets-v1/atomic-facet/atomic-facet.tsx

```diff
@@ -184,43 +188,85 @@ export class AtomicFacet
   }
 
   private renderValue(facetValue: FacetValue, onClick: () => void) {
+    const displayValue = this.bindings.i18n.t(facetValue.value);
+    const isSelected = facetValue.state === 'selected';
     switch (this.displayValuesAs) {
       case 'checkbox':
         return (
           <FacetValueCheckbox
-            value={facetValue.value}
+            displayValue={displayValue}
             numberOfResults={facetValue.numberOfResults}
-            state={facetValue.state}
+            isSelected={isSelected}
             i18n={this.bindings.i18n}
             onClick={onClick}
           ></FacetValueCheckbox>
         );
       case 'link':
-        return <FacetValueLink value={facetValue.value}></FacetValueLink>;
+        return (
+          <FacetValueLink
+            displayValue={displayValue}
+            numberOfResults={facetValue.numberOfResults}
+            isSelected={isSelected}
+            i18n={this.bindings.i18n}
+            onClick={() => {
+              this.deselectValues();
+              onClick();
+            }}
+          ></FacetValueLink>
+        );
       case 'box':
-        return <FacetValueBox value={facetValue.value}></FacetValueBox>;
+        return (
+          <FacetValueBox
+            displayValue={displayValue}
+            numberOfResults={facetValue.numberOfResults}
+            isSelected={isSelected}
+            i18n={this.bindings.i18n}
+            onClick={onClick}
```

Comment:
How com the `onClick` for the box facets is different than the one for the link facets?

Reply:
Is it because the link is only single select while box can be multi-select?

Reply:
On the link facet, since it's single select, we have to set all selected values to idle before selecting the right value.

Reply:
We could (should?) create this functionality in headless facet controller.

It's been asked by at least one client so far, and while they were able to figure out how to do it (dispatching action like you are doing here), it could still be nice to make it easier for everyone.

Reply:
Or perhaps a better solution would be to have the singleSelect be an option on the facet, more than 1 method would benefit (selecting search results/selecting value, etc)

Reply:
I'm cautious of adding methods and options, exactly because of these cross-interactions which often lead to complexity and bugs. Keep in mind that flexibility around UX is a key reason why headless exposes its actions.

If we expect clients to ask for single-select often, then it's worth considering supporting. A pure DIY sequence is quite complex:
1. deselectAllFacetValues
2. toggleSelectFacetValue
3. freezeValues
4. executeSearch(// either logFacetSelect or logFacetDeselect)

But one could also just do:
1. deselectAllFacetValues
2. controller.toggleSelect(value) // handles steps 2,3,4 above

I find composing an action with a controller to be a more scalable solution. It's an instinct I want to see developers adopt, versus us needing to provide every UX combination.

On the other hand, if we decide to make single select part of headless, then let's do it consistently to avoid questions as to why it is provided for one but not the other. This would include the facet, date facet and numeric facet controllers (category is only single-select),

Reply:
> But one could also just do

It's a bit more complex as explained in the other comment https://github.com/coveo/ui-kit/pull/874/files#r648619852

We also have to think about the search `select` method

Reply:
I agree that this would be an entire new feature for all the facets and require it's own story/PR, perhaps we should wait if the request is more and more frequent